### PR TITLE
Update reqwest to 0.9.5

### PR DESCRIPTION
**Pull request by jonas-schievink**
This prevents https://github.com/sfackler/rust-openssl/issues/994

### CODE REVIEW

**Comment by jonas-schievink** [comment]
CI fails due to rustfmt upgrade, let me know if I should run `cargo fmt`

**Comment by breeswish** [comment]
Hmmm. Interesting. Stable failed but nightly succeeded, which means there will be different code format for the same code under different versions.. @overvenus What's your idea?

**Comment by kennytm** [comment]
@breeswish rustfmt is only executed on stable.

**Comment by breeswish** [comment]
@kennytm Oh! sorry. I forgot that!

@jonas-schievink So maybe you can just run rustfmt to keep track with latest format spec (and pass CI 🤣).

**Comment by kennytm** [comment]
I suggest we fix the warnings for the two lints in `lib.rs`, and add `#[cfg-attr(..., allow(clippy::renamed_and_removed_lints))]` onto

https://github.com/pingcap/rust-prometheus/blob/2742561b84b70f8a623effd834d31c7bc5ce6bbd/src/lib.rs#L55-L57

until stepancheg/rust-protobuf#331 is resolved.

**Comment by jonas-schievink** [comment]
Any `#[allow]` on the `pub mod proto;` seems to just be ignored

File: src/lib.rs [hunk 1]
```diff
@@ -107,10 +107,7 @@ This library supports four features:
 
 */
 
-#![cfg_attr(
-    feature = "cargo-clippy",
-    allow(needless_pass_by_value, new_without_default_derive)
-)]
+#![allow(clippy::needless_pass_by_value, clippy::new_without_default_derive)]
```
[the remark this answers is not in the document]
Comment (jonas-schievink) [reply src/lib.rs]: I can add it back, but `feature = "cargo-clippy"` does not seem to exist anywhere (CI also doesn't set it)

[the remark this answers is not in the document]
Comment (jonas-schievink) [reply src/lib.rs]: Ah never mind, clippy sets that automatically

[the remark this answers is not in the document]
Comment (kennytm) [reply src/lib.rs]: @jonas-schievink this also applies to the `rustfmt::skip` attribute BTW.

**Comment by siddontang** [comment]
Ping @breeswish

**Pull request merged**